internal/controller/appconfigurationprovider_controller_test.go

```diff
@@ -713,7 +718,37 @@ var _ = Describe("AppConfiguationProvider controller", func() {
 				},
 			}
 
-			mockConfigurationSettings.EXPECT().ResolveSecretReferences(gomock.Any(), gomock.Any(), gomock.Any()).Return(newResolvedSecret, nil)
+			newSecretMetadata := make(map[string]loader.KeyVaultSecretMetadata)
+			newSecretMetadata["testSecretKey"] = loader.KeyVaultSecretMetadata{
+				HostName: "fakeHostName",
+			}
+			mockedSecretReference := make(map[string]*loader.TargetSecretReference)
+			mockedSecretReference[secretName] = &loader.TargetSecretReference{
+				Type:           corev1.SecretType("Opaque"),
+				SecretMetadata: newSecretMetadata,
+			}
+
+			newTargetSettings := &loader.TargetKeyValueSettings{
+				SecretSettings:   newResolvedSecret,
+				SecretReferences: mockedSecretReference,
+			}
+
+			mockConfigurationSettings.EXPECT().ResolveSecretReferences(gomock.Any(), gomock.Any(), gomock.Any()).Return(newTargetSettings, nil)
```

Comment:
I didn't quite sure why check the refresh two times. And seems you didn't add the mocks for the secretId, how it can be refreshed without checking secretID?

Reply:
This test covers 2 refresh scenarios, the first time refresh when secretMetaData change, the second time not refresh when data not change. The first time refresh because mocked secretReferences length changed. Will add secretID.

Reply:
Got you, and please add the scenario for the secretId check